### DISCUSSION TITLE
NAT with Postal gets be blocked on SpamHouse

### DISCUSSION THREAD

**Discussion by CodeSkills** [postalserver/postal#2688] [Help with using Postal]
Hey there,

For the last month and a half, I have been trying to solve an issue when I was getting blocked on https://check.spamhaus.org/listed/?searchterm=77.78.86.181

You can see the issue there. I blocked everything outgoing from my network and kept just Postal running and yet it still says something like "If this IP operates as a mail server, it should look and behave like a mail server. The HELO currently used appears to be dynamic and that is a behavior commonly observed in malware/proxy networks." 

I understand but I run this dynamically to make it more secure, since I can control the proxy and filter the traffic. 

Does anyone has any ideas how to get this fixed when Postal is running behind NAT?

**Comment by willpower232** [comment]
![image](https://github.com/postalserver/postal/assets/1619102/ef1105e2-4792-4dee-861c-e23f614a1427)

I'm going to presume it is salty because the hostname of the server includes elements of the IP address which, whilst probably valid for your naming structure, definitely match the kinds of hostnames seen on dynamic reverse hosts.

If you're able to rename your server to something less coincidentally numeric, that would probably do the trick.

**Reply by CodeSkills** [reply]
Hmm, we had previously full IP there that was getting blocked like that so we inspired ourselves like sparkpostmail and they have it the same like this format. So I didn't think it can be even the case of this. I will change the rDNS to something like mta-a and test again and update here as I get info if it resolved the issue.
